sdks/python/apache_beam/io/textio.py

```diff
@@ -86,7 +90,22 @@ def __init__(self,
                coder,
                buffer_size=DEFAULT_READ_BUFFER_SIZE,
                validate=True,
-               skip_header_lines=0):
+               skip_header_lines=0,
+               header_processor_fns=(None, None)):
+    """Initialize a _TextSource
+
+    Args:
+      header_processor_fns (tuple): a tuple of a header_matcher function
+        and a header_processor function. The header_matcher should return True
+        for all lines at the start of the file that are part of the file header
+        and False otherwise. These header_lines will be skipped over when
+        reading records and instead passed into header_processor to be handled.
```

Comment:
<!--thread_id:cc_148153577_t; commit:e00f526d58e10e80c9f0f7b77f2f5d2bd51d9df2; resolved:1-->
<!--section:context-quote-->
> **mhsaul** wrote:
> header_lines will be passed to the header_processor function. By "These header_lines will be skipped over when reading records", I meant that the _read_records function will not yield these lines. I'll change the description to make this a bit more clear.

<!--section:body-->
I think it makes sense to keep these as a single pair of arguments since it doesn't make sense to accept one without the other.